src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -6107,13 +6119,16 @@
         }
 
         /// <summary>
-        /// Adds an entry in <see cref="CompiledRegexRunner._searchValues"/> for the given <paramref name="chars"/> and emits a load of that initialized value.
+        /// Adds an entry in <see cref="CompiledRegexRunner._searchValues"/> for the given <paramref name="values"/> and emits a load of that initialized value.
         /// </summary>
-        private void LoadSearchValues(ReadOnlySpan<char> chars)
+        private void LoadSearchValues<T>(T[] values)
         {
-            List<SearchValues<char>> list = _searchValues ??= new();
+            List<object> list = _searchValues ??= new();
             int index = list.Count;
-            list.Add(SearchValues.Create(chars));
+            list.Add(
+                typeof(T) == typeof(char) ? SearchValues.Create((char[])(object)values) :
```

Comment:
Might be overkill but would it be worth it instead to store a bool which dictates which one of the two we are using so that we don't have to cast `object`?

Reply:
I don't think that's worthwhile. But thanks.